src/main/java/com/yourssu/roomescape/reservation/ReservationService.java

```diff
@@ -3,68 +3,99 @@
+        Theme theme = themeRepository.findById(reservationSaveRequest.theme())
+                .orElseThrow(() -> new CustomException(ErrorCode.THEME_NOT_FOUND));
 
         Reservation reservation = new Reservation(
-                memberName,
-                reservationRequest.getDate(),
+                existingMember,
+                reservationSaveRequest.date(),
                 time,
-                theme
+                theme,
+                reservationSaveRequest.status()
         );
 
-        Reservation newReservation = reservationDao.save(reservation);
+        Reservation newReservation = reservationRepository.save(reservation);
 
-        return new ReservationResponse(
+        return new ReservationSaveResponse(
                 newReservation.getId(),
-                newReservation.getName(),
+                newReservation.getMember().getName(),
                 newReservation.getTheme().getName(),
                 newReservation.getDate(),
-                newReservation.getTime().getValue());
+                newReservation.getTime().getValue(),
+                newReservation.getStatus()
+        );
     }
 
+    @Transactional
     public void deleteById(Long id) {
-        reservationDao.deleteById(id);
+        reservationRepository.deleteById(id);
     }
 
-    public List<ReservationResponse> findAll() {
-        return reservationDao.findAll().stream()
-                .map(it -> new ReservationResponse(it.getId(), it.getName(), it.getTheme().getName(), it.getDate(), it.getTime().getValue()))
-                .toList();
+    public List<ReservationFindAllResponse> getMyReservations(Member member) {
+        List<Reservation> reservedReservations = reservationRepository.findByMemberAndStatus(member, ReservationStatus.RESERVED);
+        List<ReservationWaitingWithRank> waitingsWithRank = reservationRepository.findWaitingsWithRankByMember(member);
+
+        return Stream.concat(
+                        reservedReservations.stream().map(reservation -> new ReservationFindAllResponse(
+                                reservation.getId(),
+                                reservation.getTheme().getName(),
+                                reservation.getDate(),
+                                reservation.getTime().getValue(),
+                                reservation.getStatus().getDescription()
+                        )),
+                        waitingsWithRank.stream().map(waitingWithRank -> new ReservationFindAllResponse(
+                                waitingWithRank.reservation().getId(),
+                                waitingWithRank.reservation().getTheme().getName(),
+                                waitingWithRank.reservation().getDate(),
+                                waitingWithRank.reservation().getTime().getValue(),
+                                (waitingWithRank.rank() + 1) + "번째 예약대기"
+                        ))
+                )
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                )
                .toList();
```

이렇게 작성할 수도 있어요. 둘의 차이가 무엇일까요?

Reply:
핵심은 리턴되는 List의 수정 가능 여부와 null 허용 여부에 있습니다.

 Java 8에서는 collect(Collectors.toList())를 사용하면 수정 가능한 List가 반환됩니다. null 값도 허용됩니다.
Java 10에서 도입된 toUnmodifiableList()는 수정 불가능한 List를 반환합니다. 그러나 이 방식은 null 값을 허용하지 않기 때문에, null을 삽입하려고 하면 NullPointerException이 발생할 수 있습니다.
Java 16에서는 Stream.toList()가 도입되었으며, 이는 내부적으로 toUnmodifiableList()와 같은 기능을 제공하면서, 수정 불가능한 List를 반환합니다. 또한, null 값도 허용합니다. 이름이 간단해져 코드 작성이 더 간결해졌습니다.

위 코드에서는 Stream.concat()으로 결합한 두 스트림을 하나의 리스트로 반환하며 수정이 없기 때문에 toList()가 적절해보이네요.